clang/include/clang/Basic/DiagnosticLexKinds.td

```diff
@@ -932,6 +932,12 @@ def warn_module_conflict : Warning<
   InGroup<ModuleConflict>;
 
 // C++20 modules
+def err_module_decl_cannot_be_macros : Error<
+  "the module name in a module%select{| partition}0 declaration cannot contain "
+  "an object-like macro %1, it's consists of one or more identifiers "
+  "separated by '.'">;
```

Comment:
```suggestion
  "an object-like macro %1">;
```

Reply:
Agree, I want to add a more help information into diagnostic to address @Bigcheese 's comments before. I don’t know if we need to add a link to clang’s documentation(or other helpful informations). I don’t know if there is any precedent before.

> It would be nice in cases like this if we just linked to some documentation saying what to do instead.